File: Lab02/python/CiurulEratostene/CiurulEratostene/CiurulEratostene.py
'''
4.Implement the sieve of Eratosthenes algorithm for generating all prime numbers less than a
given bound
'''
def CiurulEratostene(n): 
      
    #initialize with true all the numbers in range 1 to n
    prime = [True for i in range(n+1)] 
    p = 2
    #have a p to go from 2 to p square
    while (p * p <= n): 
        # If prime[p] is tur, then it is a prime 
        if (prime[p] == True): 
            # Giving the false value to all multiples of p starting from p*p,with the step p
            for i in range(p * p, n+1, p): 
                prime[i] = False
        #incrementing p
        p += 1
      
    # Print all prime numbers 
    for p in range(2, n): 
        if prime[p]: 
            print(p) 
  
if __name__=='__main__': 
    #input the number
    while True:
        n =  input("Give your n =  ")
        #validating the input
        if n and n.isdigit():
            print("The result is")
            CiurulEratostene(int(n))
        else:
            print("Invalid input")

File: Lab02/CiurulEratostene/CiurulEratostene/CiurulEratostene.cpp
// CiurulEratostene.cpp : Defines the entry point for the console application.
//
#include "stdafx.h"
#include <vector>
#include "CiurulEratostene.h"
#include<stdlib.h>
#include<iostream>
using namespace std;

//main function that computes all
void ciurulEratostente(long long n)
{
	//the dynamic list with all the numbers 
	vector<long long> prime;
	//initializing the list ,
	//  A value in prime[i] will 
	// finally be false if i is Not a prime, else true. 
	for (long i = 1; i <= n; i++)
	{
		prime.push_back(i);
	}

	//starting from the first prime number (which is 2)
	for (long long p = 2; p*p <= n; p++)
	{
		// If prime[p] is the same number, then it is a prime 
		if (prime[p] == p)
		{
			// Update all multiples of p (we put -1) greater than or  
			// equal to the square of it 
			// numbers which are multiple of p and are 
			// less than p^2 are already been marked.  
			for (int i = p * p; i <= n; i += p)
				prime[i] = -1;
		}

	}

	// Print all prime numbers 
	for (int p = 2; p < n; p++)
		if (prime[p] !=-1 )
			cout << p << " ";

}


int main()
{
	long long  n = 30;
	cout << "Prime numbers less than " << n << " are " << endl;
	ciurulEratostente(n);
	return 0;


	return 0;
}

